python/ray/tune/ray_trial_executor.py

```diff
@@ -370,11 +377,14 @@ def _update_avail_resources(self, num_retries=5):
 
         self._avail_resources = Resources(
             int(num_cpus), int(num_gpus), custom_resources=custom_resources)
+        self._last_resource_refresh = time.time()
         self._resources_initialized = True
 
     def has_resources(self, resources):
         """Returns whether this runner has at least the specified resources."""
-        self._update_avail_resources()
+        if time.time() - self._last_resource_refresh > self._refresh_period:
```

Comment:
Does this mean that if my trials run really fast (say 10ms) I won't be able to schedule more than 2 a second?

Why not refresh after each event processed?

Reply:
Oh, IIUC this is just for the "cluster resources", which are not expected to change dynamically often. If that's right then it seems ok, but I would document this.